### PR TITLE
Update to last rustc version

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/gtk/widgets/info_bar.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ use glib::translate::ToGlibPtr;
 use gtk::MessageType;
 use gtk::cast::GTK_INFOBAR;
 use gtk::{self, ffi};
-use glib::{to_bool, to_gboolean};
+use glib::to_gboolean;
```
Comment (gkoz) [inline src/gtk/widgets/info_bar.rs]: Hehe, you too have fallen for removing imports that aren't needed with GTK_3_6 ;)
I'm thinking, maybe we should gather the conversion stuff up into a crate-private prelude that gets included everywhere by glob (there's an RFC to make it automatic).

File: src/gtk/widgets/recent_filter_info.rs [hunk 2]
```diff
@@ -76,18 +76,18 @@ impl RecentFilterInfo {
         let mut t_groups = Vec::with_capacity(self.groups.len());
 
         for tmp in self.applications.iter() {
-            let c = CString::from_slice(tmp.as_bytes());
+            let c = CString::new(tmp.as_bytes()).unwrap();
```
Comment (gkoz) [inline src/gtk/widgets/recent_filter_info.rs]: This function still suffers from the classic dangling pointer syndrome... I still think this module needs more work than it is worth.

File: src/gtk/widgets/info_bar.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/gtk/widgets/info_bar.rs]: It was too much for me haha.

File: src/gtk/widgets/recent_filter_info.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/gtk/widgets/recent_filter_info.rs]: The point here was to remove warnings by updating the code, not to work on the code specifically. But yes, it does need it.

**Pull request merged**